components/menu/index.css

```diff
@@ -10,170 +10,676 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+  }
+  
+  .spectrum-Menu-itemIcon {
+    fill: var(--highcontrast-menu-item-label-icon-color-disabled, var(--mod-menu-item-label-icon-color-disabled, var(--spectrum-menu-item-label-icon-color-disabled)));
   }
 
-  &.is-selected {
-    /* Redundant, but included for backwards compatibility */
-    padding-inline-end: calc(
-      var(--spectrum-listitem-texticon-padding-right) -
-        var(--spectrum-popover-border-size)
-    );
+  &:hover {
+    cursor: default;
 
+    .spectrum-Menu-itemIcon {
+      fill: var(--highcontrast-menu-item-label-icon-color-disabled, var(--mod-menu-item-label-icon-color-disabled, var(--spectrum-menu-item-label-icon-color-disabled)));
+    }
+  }
+}
+
+.spectrum-Menu-itemIcon {
+  grid-area: iconArea;
+  align-self: center;
+}
+
+.spectrum-Menu-checkmark {
+  grid-area: checkmarkArea;
+}
+
+/* menus with "selectable" menuitems */
+.spectrum-Menu.is-selectable {
+  /* provide inline-start spacing for potential checkmarks */
+  .spectrum-Menu-item {
+    padding-inline-start: var(--spectrum-menu-item-selectable-edge-to-text-not-selected);
+  }
+
+  .spectrum-Menu-checkmark {
+    display: none;
+  }
+
+  /* remove inline-start spacing once an item is selected and a checkmark appears */
+  .spectrum-Menu-item.is-selected {
+    padding-inline-start: var(--spectrum-menu-item-label-inline-edge-to-content);
+    
     .spectrum-Menu-checkmark {
       display: block;
     }
   }
+}
 
-  .spectrum-Icon,
-  .spectrum-Menu-itemIcon {
-    /* Don't get smaller, you're an icon! */
-    flex-shrink: 0;
-    align-self: flex-start;
-  }
+.spectrum-menu-itemSelection {
+  grid-area: selectedArea;
+}
+
+.spectrum-Menu-itemLabel {
+  grid-area: labelArea;
+  font-size: var(--mod-menu-item-label-font-size, var(--spectrum-menu-item-label-font-size));
+  color: var(--highcontrast-menu-item-label-content-color-default, var(--mod-menu-item-label-content-color-default, var(--spectrum-menu-item-label-content-color-default)));
+}
 
-  .spectrum-Icon + .spectrum-Menu-itemLabel,
-  .spectrum-Menu-itemIcon + .spectrum-Menu-itemLabel {
-    margin-inline-start: var(--spectrum-listitem-texticon-icon-gap);
+.spectrum-Menu-itemValue {
+  grid-area: valueArea;
+}
+
+.spectrum-Menu-itemActions {
+  grid-area: actionsArea;
+}
 
-    inline-size: calc(
-      100% - var(--spectrum-listitem-texticon-ui-icon-width) -
-        var(--spectrum-listitem-texticon-icon-gap) -
-        var(--spectrum-listitem-textthumbnail-padding-left) -
-        var(--spectrum-alias-workflow-icon-size-m)
-    );
+.spectrum-Menu-chevron {
+  grid-area: chevronArea;
+  align-self: center;
+  block-size: var(--spectrum-menu-item-checkmark-height);
+  height: var(--spectrum-menu-item-checkmark-height);
+  inline-size: var(--spectrum-menu-item-checkmark-width);
+  width: var(--spectrum-menu-item-checkmark-width);
+}
+
+/* TODO: double check this. Is this necessary? An error? */
```

Comment:
What may not be necessary?

Reply:
I think I left this in from an earlier commit when I was taking notes for myself. Removed!